confuse/util.py

```diff
@@ -129,14 +129,20 @@ def find_package_path(name):
     return os.path.dirname(os.path.abspath(filepath))
 
 
+def xdg_config_home():
+    """Return the value of the XDG_CONFIG_HOME
+    environment variable if it is set and nonempty,
+    and UNIX_DIR_FALLBACK otherwise
+    """
+    config_path = os.getenv('XDG_CONFIG_HOME', UNIX_DIR_FALLBACK)
+    return config_path if config_path != '' else UNIX_DIR_FALLBACK
+
 def xdg_config_dirs():
     """Returns a list of paths taken from the XDG_CONFIG_DIRS
-    and XDG_CONFIG_HOME environment varibables if they exist
+    environment variable if it exists and is nonempty
     """
     paths = []
-    if 'XDG_CONFIG_HOME' in os.environ:
-        paths.append(os.environ['XDG_CONFIG_HOME'])
-    if 'XDG_CONFIG_DIRS' in os.environ:
+    if 'XDG_CONFIG_DIRS' in os.environ and os.environ['XDG_CONFIG_DIRS'] != '':
```

Comment:
No need to look before leap here, just use the default fallback behavior (will return `None` if not present and empty string is falsy):
```suggestion
    if os.environ.get('XDG_CONFIG_DIRS'):
```